packages/ses/src/compartment-shim.js

```diff
@@ -120,6 +202,58 @@ export const CompartmentPrototype = {
   __isKnownScopeProxy__(value) {
     return privateFields.get(this).knownScopeProxies.has(value);
   },
+
+  module(specifier) {
+    if (typeof specifier !== 'string') {
+      throw new TypeError('first argument of module() must be a string');
+    }
+
+    assertModuleHooks(this);
+
+    const { exportsProxy } = getDeferredExports(
+      this,
+      privateFields.get(this),
+      moduleAliases,
+      specifier,
+    );
+
+    return exportsProxy;
+  },
+
+  async import(specifier) {
+    if (typeof specifier !== 'string') {
+      throw new TypeError('first argument of import() must be a string');
+    }
+
+    assertModuleHooks(this);
+
+    return load(privateFields, moduleAliases, this, specifier).then(() => {
+      const namespace = this.importNow(specifier);
+      return { namespace };
+    });
+  },
+
+  async load(specifier) {
+    if (typeof specifier !== 'string') {
+      throw new TypeError('first argument of load() must be a string');
+    }
+
+    assertModuleHooks(this);
+
+    return load(privateFields, moduleAliases, this, specifier);
+  },
+
+  importNow(specifier) {
+    if (typeof specifier !== 'string') {
+      throw new TypeError('first argument of importNow() must be a string');
+    }
+
+    assertModuleHooks(this);
+
+    const moduleInstance = link(privateFields, moduleAliases, this, specifier);
```

Comment:
If we `importNow` too early, where does it fail in `link`? Does it produce an understandable error?

Reply:
In `src/module-link.js`, you would see this `ReferenceError` if you attempted `importNow` before a dependency was loaded.

```js
  if (moduleRecord === undefined) {
    throw new ReferenceError(`Missing link to module ${q(moduleSpecifier)}`);
  }
```

Reply:
Ah of course. The missing link!